elastic/kibana
# Integrate painless autocomplete in runtime fields editor

This PR integrates painless autocompletion in the runtime fields editor.

Continuation of https://github.com/elastic/kibana/pull/80577.

### How to test
Verify autocomplete functionality is working in the runtime field editor. The runtime field editor is currently only integrated in the composable index template, legacy index template, and component template wizards.

For a list of available autocomplete functionality, check out the "Testing" section in https://github.com/elastic/kibana/pull/80577.

In order to test field autocompletion, you need to create at least one mapped field.

### Known issues
I've received some feedback already around improvements that can be made to the autocomplete. I will handle these in a separate PR (https://github.com/elastic/kibana/pull/85055):
- Auto closing brackets
- Autocomplete support for `emit()` and `params`
- Filter out `text` field types from the field autocomplete suggestions; `text` field types are never in doc values.
- Autocomplete should not kick in for string and boolean values
